### PR TITLE
glob method && Implement FTPS implicit FTP over TLS

### PR DESCRIPTION
**Pull request by catalint**
## glob method
ported Glob method from filepath - https://golang.org/pkg/path/filepath/#Glob and also present in sftp client http://godoc.org/github.com/pkg/sftp#Client.Glob

## Implement FTPS implicit FTP over TLS
This will allow the package to also be able to connect to FTPS on servers that have implicit tls enabled
added new method func DialImplicitTLS(addr string, config *tls.Config) (*ServerConn, error)
added new TLS server and ran tests over TLS also, all pass

## increase test coverage

### CODE REVIEW

**Comment by catalint** [comment]
travis failed because one of the tests error-ed with 
```
Bad response status from coveralls: 422
{"message":"Couldn't find a repository matching this job.","error":true}
```

this pull request is good for you to look at
thanks

**Comment by corny** [comment]
Please do two separate pull requests - one for each feature.

**Comment by catalint** [comment]
The commits are different, do you really need 2 pull requests?

**Comment by corny** [comment]
Yes please. That makes it easier for reviewing. I don't think your Glob-function can be merged in the current state. What are the license requirements for copying code from the Go standard library?

File: client_test.go [hunk 1]
```diff
@@ -14,20 +16,39 @@ const (
 	testDir  = "mydir"
 )
 
+func isTLSServer() bool {
+	return os.Getenv("FTP_SERVER") == "vsftpd_implicit_tls"
+}
+
+func getConnection() (*ServerConn, error) {
+	if isTLSServer() {
+		return DialImplicitTLS("localhost:21", &tls.Config{InsecureSkipVerify: true})
```
Comment (tomtitchener) [inline client_test.go]: Why is that port 21 and not 990?

**Comment by corny** [comment]
I checked out your commits and I read the specification of implicit TLS for FTP. Your changes secure the control channel but not the data channel. Are there seriously people still preffering FTP/S to SCP and SFTP?

I don't think it worths the effort:
https://tools.ietf.org/html/rfc4217#section-9

**Comment by garymoon** [comment]
As an example of not-by-choice, Azure (also not by choice) supports FTP access to their "Azure Websites" feature. It (horrifyingly) supports plain FTP, but also supports FTPS/FTPES which I would much prefer to use.

Thanks so much for working on support for this everyone!

P.S. I found this thread via the rclone forums, it would be awesome to have FTPS/FTPES support in there. Really cool of you to be following up personally also @ncw.

**Pull request closed without merging**